### ISSUE TITLE
Listing a bucket with about 200 objects gets stuck

### ISSUE THREAD

**Issue by andriytk** [andriytk/ceph-old#10]
When the bucket has about 200 objects, listing it just gets stuck:

```
s3cmd --no-ssl ls s3://gttest3
```

Note, that running it with `--debug` option shows the dump of objects immediately.

**Comment by bkirunge7** [comment]
radosgw get stuck at below whne list is running on bucket with atleast 200 objects

rgw::auth::SysReqApplier -> rgw::auth::LocalApplier(acct_user=tenant$ns$tester, acct_name=Motr Explorer, subuser=, perm_mask=15, is_admin=0), owner=tenant$ns$tester, perm=1
2021-12-03T09:10:32.454+0000 7f7665b3b700 10 req 0 0.005000035s s3:list_bucket  identity=rgw::auth::SysReqApplier -> rgw::auth::LocalApplier(acct_user=tenant$ns$tester, acct_name=Motr Explorer, subuser=, perm_mask=15, is_admin=0) requested perm (type)=1, policy perm=1, user_perm_mask=1, acl perm=1
2021-12-03T09:10:32.454+0000 7f7665b3b700  2 req 0 0.005000035s s3:list_bucket verifying op params
2021-12-03T09:10:32.454+0000 7f7665b3b700  2 req 0 0.005000035s s3:list_bucket pre-executing
2021-12-03T09:10:32.454+0000 7f7665b3b700  2 req 0 0.005000035s s3:list_bucket executing
2021-12-03T09:10:32.454+0000 7f7665b3b700  0 id = 0x 9f931fbb205496e9:0x26fd305233e2c25d
2021-12-03T09:10:32.454+0000 7f7665b3b700  0 converted id = 0x 78000000205496e9:0x26fd305233e2c25d
2021-12-03T09:10:32.455+0000 7f7665b3b700  0 index name  = motr.rgw.bucket.index.test1
2021-12-03T09:10:32.455+0000 7f7665b3b700  0 key.data address  =
2021-12-03T09:10:32.455+0000 7f7665b3b700  0 bv->ov_buf[0] = 0
2021-12-03T09:10:32.455+0000 7f7665b3b700  0 bv->ov_vec.v_count[0]0

**Comment by andriytk** [comment]
Fixed at #12:
```
$ s3cmd --no-ssl ls s3://testbucket1/ | wc -l
1002
$ s3cmd --no-ssl ls s3://testbucket1/a | wc -l
69
```
